packages/next/src/client/components/react-dev-overlay/server/middleware-webpack.ts

```diff
@@ -106,8 +109,9 @@ async function findOriginalSourcePositionAndContent(
 
   try {
     const sourcePosition = consumer.originalPositionFor({
-      line: position.line,
-      column: position.column ?? 0,
+      line: position.lineNumber ?? 1,
+      // 0-based columns out requires 0-based columns in.
+      column: (position.column ?? 1) - 1,
```

Comment:
This is the actual fix. The rest just makes this middleware more similar to the Turbopack middleware while also creating less intermediate objects.